Redesign editor setup screen with new "form" controls

bdach opened the pull request:
- [x] Depends on https://github.com/ppy/osu/pull/30098
- Closes https://github.com/ppy/osu/issues/29643

https://github.com/user-attachments/assets/8c17219d-491a-4951-a9c3-fc77b7662566

Also contains some assorted drive-by fixes. See commit messages for details.

bdach committed:
Replace setup screen controls with new "form" controls

bdach committed:
Fix "form" file selector displaying commit animation on initial show

bdach committed:
Fix "form" textbox not dropping border if disabled when hovered

bdach committed:
Autoselect contents of "form" number box

peppy committed:
Merge branch 'master' into editor/setup-screen

bdach committed:
Fix code quality

bdach committed:
Fix tests

peppy commented:
I think metadata deserves to be above resources

```diff
diff --git a/osu.Game/Screens/Edit/Setup/SetupScreen.cs b/osu.Game/Screens/Edit/Setup/SetupScreen.cs
index 4b9a7a858f..16d58efb93 100644
--- a/osu.Game/Screens/Edit/Setup/SetupScreen.cs
+++ b/osu.Game/Screens/Edit/Setup/SetupScreen.cs
@@ -72,8 +72,8 @@ private void load(EditorBeatmap beatmap, OverlayColourProvider colourProvider)
                                             Spacing = new Vector2(25),
                                             Children = new Drawable[]
                                             {
-                                                new ResourcesSection(),
                                                 new MetadataSection(),
+                                                new ResourcesSection(),
                                             }
                                         },
                                         new FillFlowContainer

```

bdach committed:
Redo the layout of sections based on discord feedback

See
https://discord.com/channels/188630481301012481/188630652340404224/1291358770064130140
and everything after.

bdach committed:
Adjust slider bar padding

https://discord.com/channels/188630481301012481/188630652340404224/1291374650256916482

bdach committed:
Make colour palette things circular again

bdach committed:
Ensure columns collapse into one correctly if no space

bdach committed:
Fix tests again

peppy reviewed (approved):
Feels good enough to push forward
